app/src/main/java/com/futurice/freesound/feature/search/DefaultSearchDataModel.java

```diff
@@ -93,19 +97,19 @@ public Completable clear() {
 
     private void storeValueAndClearError(@NonNull final Option<List<Sound>> listOption) {
         lastResultsOnceAndStream.onNext(listOption);
-        lastErrorOnceAndStream.onNext(Option.none());
+        searchStateOnceAndStream.onNext(SEARCH_CLEARED());
```

Comment:
I think this reads much better than before - it's more obvious what the event is.